### PR TITLE
Ignore empty directories

### LINKED ISSUE #51: Add option to drop empty folders from the result when recursive is set to "true"
Currently if there are subfolders in the directory, but there are no files matching the filter, empty object for the subfolder still gets added to the results, even though it is just a noise in such case.

### PR DESCRIPTION
**Pull request by kibertoad**
fixes #51

### CODE REVIEW

File: Changes.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ This file is a manually maintained list of changes for each release. Feel free
 to add your changes here when sending pull requests. Also send corrections if
 you spot any mistakes.
 
+## Unreleased
```
Comment (dougwilson) [inline Changes.md]: I appreciate taking the time to do this particular change. If you want to, then can we match all previous commits? https://github.com/felixge/node-require-all/commit/a4efa8b4eaace1bf1ad4ce102a1a705925ac1363 for instance.

File: index.js [hunk 2]
```diff
@@ -37,13 +37,18 @@ module.exports = function requireAll(options) {
...
 
-      modules[map(file, filepath)] = requireAll({
-        dirname: filepath,
-        filter: filter,
-        excludeDirs: excludeDirs,
-        map: map,
-        resolve: resolve
-      });
+      var directoryModules = requireAll({
+		  dirname: filepath,
```
Comment (dougwilson) [inline index.js]: It seems like the intention is messed up in the changes. GitHub is showing the new line significantly more indented than before.

Comment (kibertoad) [reply index.js]: sorry, my bad.

File: Changes.md (hunk 1 above)
Comment (kibertoad) [reply Changes.md]: done

**Comment by dougwilson** [comment]
I have been very ill and haven't had the chance to folow up yet. I apologize. Just leaving the dr office now.

**Comment by kibertoad** [comment]
Sorry to hear that. Please get better!

**Comment by kibertoad** [comment]
@dougwilson How are you feeling these days :)?

**Comment by dougwilson** [comment]
Just starting to pick everything back up now :+1: your changes to this module have been top of mind, on fact. Was going to get them last night but just needed to check up on some gh conversations. Plan to get this all done today. I know that may not mean much, but I'm trying to be honest with my plans to you so you don't feel like I'm ignoring you or anything.

**Comment by kibertoad** [comment]
Thank you, I appreciate it ;)

**Comment by dougwilson** [comment]
Changes look good, but the changeset doesn't actually include the empty dirs at all that the tests are asserting against. When I clone the branch, this is all I see in the withempty dir:
```
$ ls -R test/withempty
subdir

test/withempty/subdir:
subdir2

test/withempty/subdir/subdir2:
main-Controller.js
```

This actually causes the new test added to pass against the current `master`, which means the test isn't actually asserting that empty dirs were actually ignored. I just started taking a look. I may end up deferring looking into this issue to tomorrow 👍

**Comment by kibertoad** [comment]
Probably I should add code to create them programmatically :)

**Comment by dougwilson** [comment]
Sorry, I have been out and I am helping out @howtoclient today so I haven't been able to get to your issue yet today. I may need to wait until tomorrow, but definitely plan to get the tests fixed for you soon :+1:

**Comment by kibertoad** [comment]
@dougwilson Sorry for not getting to this earlier; were test just dropped? Can't find them now.

**Pull request merged**